presto-main/src/main/java/com/facebook/presto/execution/AccessControlCheckerExecution.java

```diff
@@ -262,6 +270,10 @@ private ListenableFuture<?> executeTask()
         stateMachine.beginColumnAccessPermissionChecking();
         checkAccessPermissions(queryAnalysis.getAccessControlReferences());
         stateMachine.endColumnAccessPermissionChecking();
+        if (preparedQuery.isExplainTypeIO()) {
+            explainIOResult = ((QuerySpecification) ((Query) ((BuiltInQueryAnalysis) queryAnalysis).getAnalysis().getStatement()).getQueryBody()).getFrom()
```

Comment:
still trying to understand the logic here, but first thing that pops out: why is BuiltInQueryAnalysis specified directly here? shouldn't the abstract BuiltInQueryAnalysis have the same APIs (even if they are not implemented for some implementers)?